<launch>

  <!-- Detect Ball Parameters-->
  <param name="/detect_ball_forward/display" value="0" />   <!-- 0 will not display images -->
  <param name="/detect_ball_forward/require_correct_state" value="1" />   <!-- 1 requires the correct state -->
  
  <param name="/detect_ball_forward/h_min_orange" value="0" />
  <param name="/detect_ball_forward/h_max_orange" value="27" />
  <param name="/detect_ball_forward/s_min_orange" value="52" />
  <param name="/detect_ball_forward/s_max_orange" value="255" />
  <param name="/detect_ball_forward/v_min_orange" value="111" />
  <param name="/detect_ball_forward/v_max_orange" value="255" />

  <param name="/detect_ball_forward/h_min_green" value="24" />
  <param name="/detect_ball_forward/h_max_green" value="44" />
  <param name="/detect_ball_forward/s_min_green" value="18" />
  <param name="/detect_ball_forward/s_max_green" value="160" />
  <param name="/detect_ball_forward/v_min_green" value="148" />
  <param name="/detect_ball_forward/v_max_green" value="255" />

  <node name="detect_ball_forward" pkg="coconuts_odroid" type="detect_ball_forward" output="screen"/>

  <!-- stereo image proc - forward camera-->
  <node ns="camera_forward" name= "image_proc" pkg="image_proc" type="image_proc"/>

  <!-- forward camera uses usb_cam -->
  <node name= "camera_forward" pkg="usb_cam" type="usb_cam_node" >
    <param name="video_device" type="string" value="/dev/video0"/>  <!-- video2 for odroid -->
    <param name="pixel_format" type="string" value="yuyv"/>
    <param name="framerate" type="int" value="10"/>
    <param name="camera_frame_id" type="string" value="camera_forward" /> 
    <param name="camera_info_url" type="string" value="file://$(find coconuts_odroid)/src/camera_forward_calibration.yaml"/>
  </node>

  <!-- locate ball publishes ball transform in 3d space -->
  <node name="locate_ball_forward" pkg="coconuts_odroid" type="locate_ball_forward" output="screen" />


  </launch>
